src/Test/Perf/ConsumptionParser/Program.cs

```diff
@@ -0,0 +1,133 @@
+        {
+            const string prefix = "PerfResults-";
+            const string share = @"\\mlangfs1\public\basoundr\PerfTraces\";
+
+            var runs = new List<Tuple<string, ConsumptionParse, RunInfo>>();
+
+            // Collect run information from the share
+            foreach (var directory in Directory.GetDirectories(share))
+            {
+                var lastPath = directory.Substring(share.Length);
+                if (!lastPath.StartsWith(prefix))
+                {
+                    continue;
+                }
+
+                var date = lastPath.Substring(prefix.Length);
+                var consumptionXml = Path.Combine(directory, "ConsumptionTempResults.xml");
+                var resultJson = Directory.EnumerateFiles(directory, "Roslyn*.json").Single();
+
+                var parse = ConsumptionParse.Parse(File.ReadAllText(consumptionXml));
+                var runInfo = RunInfo.Parse(File.ReadAllText(resultJson));
+                runs.Add(Tuple.Create(date, parse, runInfo));
+
+            }
+
+            // Sort by date
+            runs.Sort((run1, run2) => run1.Item1.CompareTo(run2.Item1));
+
+            // Collect all of the metric names first 
+            var metrics = (
+                from run in runs
+                from scenario in run.Item2.Scenarios
+                from metric in scenario.Counters
+                select metric.Name
+            ).ToSet();
+
+            // Write the column titles
+            Console.Write("build, username, branch, scenario");
+            foreach (var metric in metrics)
+            {
+                Console.Write($", z_{metric}");
+            }
+            Console.WriteLine();
+
+            foreach (var run in runs)
+            {
+                var date = run.Item1;
+                var consumption = run.Item2;
+                var runInfo = run.Item3;
+
+                foreach (var scenario in consumption.Scenarios)
+                {
+                    Console.Write($"{NormalizeDate(date)}, {NormalizeUsername(runInfo.Username)}, {NormalizeBranch(runInfo.Branch)}, {NormalizeScenario(scenario.Name)}");
+                    foreach (var metric in metrics)
+                    {
+                        var m = scenario[metric];
+                        if (m != null)
+                        {
+                            var mm = m.Value;
+                            Console.Write($", {mm.Value}");
+                        }
+                        else
+                        {
+                            Console.Write(",");
+                        }
+                    }
+                    Console.WriteLine();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Removes a prefix from a string.  RemovePrefix("aaa-bbb", "aaa-") gives you "bbb".
+        /// </summary>
+        private static string RemovePrefix(string target, string prefix)
+        {
+            if (target.StartsWith(prefix))
```

Comment:
Does case matter here?  Consider using a `StringComparer` to make this explicit.